WIP: Adding support for razor API behind HTTPS

Nekototori opened the pull request:
I modified the rest call to add supported HTTPS methods along with a way to specify in the API certs etc using the existing contructs you provided.

This allows anyone running razor with an ssl cert in front of the API to use this module.

Nekototori committed:
Added HTTPS support

Nekototori committed:
Added additional support for SSL

Nekototori committed:
Removed unnecessary if

Nekototori committed:
Added SSL types to API yaml

scottsolowild commented on lib/puppet/provider/razor_rest.rb:
```diff
@@ -33,11 +33,38 @@ def self.get_rest_info
     else
       port = 8150
     end
+     if yamldata.include?('http_method')
```
Newline + indentation.

scottsolowild commented on lib/puppet/provider/razor_rest.rb:
```diff
@@ -110,10 +152,10 @@ def self.get_json_from_url(url)
     version
   end
-end
+end
```
Newline here disappeared.

scottsolowild commented on manifests/api.pp:
```diff
@@ -16,6 +16,10 @@
 class razor::api (
   String $hostname = 'localhost',
   Variant[Undef, Integer] $port = undef,
+  Variant[String, Enum['http', 'https']] $http_method = 'http',
```
These should be added to the documentation above.

natemccurdy replied:
@Nekototori Should this parameter be only one of `http` and `https`?

If so, you don't need to Variant String stuff, and can do just `Enum['http', 'https']`.

scottsolowild commented on lib/puppet/provider/razor_rest.rb:
```diff
@@ -81,10 +108,17 @@ def post_command(command, resourceHash)
+
     begin
-      RestClient.post url, resourceHash.to_json, :content_type => :json
+#      RestClient.post url, resourceHash.to_json, :content_type => :json
```
Need this?

scottsolowild commented on lib/puppet/provider/razor_rest.rb:
```diff
@@ -33,11 +33,38 @@ def self.get_rest_info
+    if yamldata.include?('client_cert')
+      client_cert = yamldata['client_cert']
+    else
+      client_cert = "/etc/puppetlabs/puppet/ssl/certs/#{hostname}.pem"
```
I think these paths should default to `nil`. The installation module should be responsible for helping this module locate these files. I suggest adding these as module parameters in `init.pp`.

Nekototori replied:
Mostly just following the existing code logic to maintain consistency.

Nekototori committed:
Prevent fact from generating errors without client installed
The fact tries to use the razor client to query the server without
verifying the client is installed.  This issue is present on any box
that happens to be using port 8080 in any fashion.  Wrapped it in an if
statement that checks for presence of the razor client first.

Also removed all sorts of trailing spaces.

Nekototori committed:
Merge branch 'master' of https://github.com/Nekototori/puppet-razor

Nekototori committed:
Partial cleanup/fixes

Nekototori committed:
Fixed broken syntax

Nekototori committed:
Removed unnecessary Variant String

Nekototori committed:
Merge branch 'master' into master

Nekototori committed:
fixing trailing space that was leftover to pass tests
